### PR TITLE
Fix potential Null pointer dereference in ws-chat-server.c

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: sample/ws-chat-server.c [hunk 1]
```diff
@@ -233,6 +233,9 @@ main(int argc, char **argv)
 	base = event_base_new();
 
 	sig_int = evsignal_new(base, SIGINT, signal_cb, base);
+	if (sig_int == NULL) {
```
Comment (azat) [inline sample/ws-chat-server.c:236]: Should clean the base in this case, i.e. `goto cleanup`

```
cleanup:
    if (base)
        event_base_free(base);
```

And should check the base then as well

**Review by azat — CHANGES_REQUESTED** [review]

File: sample/ws-chat-server.c (hunk 1 above)
Comment (icy17) [reply sample/ws-chat-server.c:236]: Got it, I'll fix it

File: sample/ws-chat-server.c [hunk 2]
```diff
@@ -255,4 +255,9 @@ main(int argc, char **argv)
 	libevent_global_shutdown();
 
 	return 0;
+
+cleanup:
+	if (base)
+	event_base_free(base);
```
Comment (azat) [inline sample/ws-chat-server.c:261]:
```suggestion
		event_base_free(base);
```

File: sample/ws-chat-server.c [hunk 3]
```diff
@@ -255,4 +255,9 @@ main(int argc, char **argv)
 	libevent_global_shutdown();
 
 	return 0;
+
+cleanup:
```
Comment (azat) [inline sample/ws-chat-server.c:259]: Can you please also add a check `http_server` and don't forget to clean the `sig_int` here

**Review by azat — CHANGES_REQUESTED** [review]

**Comment by azat** [comment]
Merged with co-authoriship - https://github.com/libevent/libevent/pull/1674

**Pull request closed without merging**